front/lib/webcrawler.ts

```diff
@@ -35,11 +35,16 @@ export function isUrlValid(url: string) {
     if (url.trim().length === 0) {
       return false;
     }
-    const u = new URL(url);
-    if (u.protocol !== "http:" && u.protocol !== "https:") {
-      return false;
-    }
-    return true;
+    const urlValidationRegex = new RegExp(
```

Comment:
Let’s avoid creating regrow everytime and extract it to the top of the file.

Reply:
:+1: moved it above function (in that particular case cleaner atop function than atop file IMHO)